src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -136,7 +134,23 @@ protected Contract toEntity(ContractDto entity) {
         contract.setEmployee(employee);
         contract.setRole(role);
         contract.setSeniority(seniority);
+        if(entity.paymentSettlement() == null) contract.setPaymentsSettlement(new HashSet<>());
 
         return contract;
     }
+
+    protected void checkActiveContract(Contract contractToCheck) {
+        Optional<Contract> activeContract = contractRepository.findContractByEmployeeIdAndActiveIsTrueAndDeletedIsFalse(contractToCheck.getEmployee().getId());
+        if ((contractToCheck.getEndDate() == null || contractToCheck.getEndDate().isAfter(LocalDate.now())) && (contractToCheck.getStartDate().isBefore(LocalDate.now())) || contractToCheck.getStartDate().isEqual(LocalDate.now()) ) {
```

Comment:
`(contractToCheck.getStartDate().isBefore(LocalDate.now())) || contractToCheck.getStartDate().isEqual(LocalDate.now())` = `!contractToCheck.getStartDate().isAfter(LocalDate.now()`

Reply:
does this conditions checks if end date is after today?